client/src/main/java/io/pravega/client/batch/impl/BatchClientFactoryImpl.java

```diff
@@ -220,6 +233,21 @@ public List<SegmentRange> getSegmentRangeBetweenStreamCuts(final StreamCut start
     public StreamCut getNextStreamCut(final StreamCut startingStreamCut, long approxDistanceToNextOffset) throws SegmentTruncatedException {
         log.debug("getNextStreamCut() -> startingStreamCut = {}, approxDistanceToNextOffset = {}", startingStreamCut, approxDistanceToNextOffset);
         Preconditions.checkArgument(approxDistanceToNextOffset > 0, "Ensure approxDistanceToNextOffset must be greater than 0");
+        return Retry.withExpBackoff(1, 10, retryAttempts, Duration.ofSeconds(30).toMillis()).retryWhen(t -> {
```

Comment:
I think it would be better to pass in the full Retry policy that way the `Retry.withExpBackoff(1, 10, 10, Duration.ofSeconds(30).toMillis())` can go into the constructor and here it can just use the member variable.